subsys/bluetooth/host/att.c

```diff
@@ -668,11 +675,23 @@ struct net_buf *bt_att_chan_create_pdu(struct bt_att_chan *chan, uint8_t op,
 		return NULL;
 	}
 
-	data = tx_meta_data_alloc(timeout);
-	if (!data) {
-		LOG_WRN("Unable to allocate ATT TX meta");
-		net_buf_unref(buf);
-		return NULL;
+	if (is_rsp) {
+		/* There can only ever be one transaction at a time on a
+		 * bearer/channel. Use a dedicated channel meta-data to ensure
+		 * we can always queue an (error) RSP for each REQ. The ATT
+		 * module can then reschedule the RSP if it is not able to send
+		 * it immediately.
+		 */
+		__ASSERT_NO_MSG(chan->transaction_meta.att_chan == NULL);
```

Comment:
Restore this invariant in the free().

Reply:
it's already being restored (by the `memset`)

Reply:
As discussed offline: This can be intentionally triggered by an remote, so it should not be an assert, but rather a disconnect or other more graceful failure.